engines/e_afalg.c

```diff
@@ -124,27 +124,59 @@ static ossl_inline int io_read(aio_context_t ctx, long n, struct iocb **iocb)
     return syscall(__NR_io_submit, ctx, n, iocb);
 }
 
+/* A version of 'struct timespec' with 32-bit time_t
+   and suseconds_t.  */
+struct __timespec32
+{
+  __kernel_long_t tv_sec;
+  __kernel_long_t tv_nsec;
+};
+
+
 static ossl_inline int io_getevents(aio_context_t ctx, long min, long max,
                                struct io_event *events,
                                struct timespec *timeout)
 {
+#if defined(__NR_io_pgetevents_time64)
+    /* Check if we are a 32-bit architecture with a 64-bit time_t */
+    if (sizeof(timeout->tv_sec) == 8 && sizeof(long) == 4) {
```

Comment:
This should check against sizeof(__kernel_long_t) in order to work on x32, same in lines 151 and 157.

As a more readable variant, you could rewrite the comparison in each case as 'sizeof(*timeout) != sizeof(struct __timespec32)'.

Reply:
Thanks, fixed.